addons/website_event_sale/controllers/main.py

```diff
@@ -23,13 +23,18 @@ def _create_attendees_from_registration_post(self, event, registration_data):
         if not any(info.get('event_ticket_id') for info in registration_data):
             return super()._create_attendees_from_registration_post(event, registration_data)
 
+        filtered_registration = self.registration_filter(registration_data)
+        if len(filtered_registration.get('free_registration_data')) == len(registration_data):
+            # skip SO creation/payment process
+            return super()._create_attendees_from_registration_post(event, registration_data)
```

Comment:
Would re-write as follows to avoid a separate method and multiple tickets browsing.

```py
event_ticket_ids = [registration['event_ticket_id'] for registration in registration_data]
event_ticket_by_id = {
    event_ticket.id: event_ticket
    for event_ticket in request.env['event.event.ticket'].sudo().browse(event_ticket_ids)
}

if all(event_ticket.price == 0 for event_ticket in event_ticket_by_id.values()):
    # all chosen tickets are free, skip SO and payment process
    return super()._create_attendees_from_registration_post(event, registration_data)
```

Then here below to build tickets data:
```py
event_ticket_id = data.get('event_ticket_id')`
event_ticket = event_ticket_by_id.get(event_ticket_id)
if event_ticket_id and event_ticket and event_ticket.price != 0:
    paid_tickets_data[event_ticket_id] += 1
```

Same thing when adding the SO reference in the other loop.
You can also modify the "ticket_sudo" reference to get the ticket from that dict to avoid a re-browse.

Do you see what I mean?

Make sure to re-test but it should do the trick.